src/data_c.c

```diff
@@ -642,6 +939,132 @@ shmem_getmem(void *target, const void *source, size_t len, int pe)
 
 
 void
+shmem_float_get_nbi(float *dest, const float *source, size_t nelems, int pe)
+{
+  SHMEM_ERR_CHECK_INITIALIZED();
+
+  shmem_internal_get(dest, source, sizeof(float)*nelems, pe);
+}
+
+
+void
+shmem_double_get_nbi(double *dest, const double *source, size_t nelems, int pe)
+{
+  SHMEM_ERR_CHECK_INITIALIZED();
+
+  shmem_internal_get(dest, source, sizeof(double)*nelems, pe);
+}
+
+
+void
+shmem_longdouble_get_nbi(long double *dest, const long double *source, size_t nelems, int pe)
+{
+  SHMEM_ERR_CHECK_INITIALIZED();
+
+  shmem_internal_get(dest, source, sizeof(long double)*nelems, pe);
+}
+
+
+void
+shmem_char_get_nbi(char *dest, const char *source, size_t nelems, int pe)
+{
+  SHMEM_ERR_CHECK_INITIALIZED();
+
+  shmem_internal_get(dest, source, sizeof(char)*nelems, pe);
+}
+
+
+void
+shmem_getmem_nbi(void *dest, const void *source, size_t nelems, int pe)
+{
+  SHMEM_ERR_CHECK_INITIALIZED();
+
+  shmem_internal_get(dest, source, nelems, pe);
+}
+
+
+void
+shmem_short_get_nbi(short *dest, const short *source, size_t nelems, int pe)
+{
+  SHMEM_ERR_CHECK_INITIALIZED();
+
+  shmem_internal_get(dest, source, sizeof(short)*nelems, pe);
+}
+
+
+void
+shmem_int_get_nbi(int *dest, const int *source, size_t nelems, int pe)
+{
+  SHMEM_ERR_CHECK_INITIALIZED();
+
+  shmem_internal_get(dest, source, sizeof(int)*nelems, pe);
+}
+
+
+void
+shmem_long_get_nbi(long *dest, const long *source, size_t nelems, int pe)
+{
+  SHMEM_ERR_CHECK_INITIALIZED();
+
+  shmem_internal_get(dest, source, sizeof(long)*nelems, pe);
+}
+
+
+void
+shmem_longlong_get_nbi(long long *dest, const long long *source, size_t nelems, int pe)
+{
+  SHMEM_ERR_CHECK_INITIALIZED();
+
+  shmem_internal_get(dest, source, sizeof(long long)*nelems, pe);
+}
+
+
+void
+shmem_get8_nbi(void *dest, const void *source, size_t nelems, int pe)
+{
+  SHMEM_ERR_CHECK_INITIALIZED();
+
+  shmem_internal_get(dest, source, nelems, pe);
+}
+
+
+void
+shmem_get16_nbi(void *dest, const void *source, size_t nelems, int pe)
+{
+  SHMEM_ERR_CHECK_INITIALIZED();
+
+  shmem_internal_get(dest, source, 2 * nelems, pe);
+}
+
+
+void
+shmem_get32_nbi(void *dest, const void *source, size_t nelems, int pe)
+{
+  SHMEM_ERR_CHECK_INITIALIZED();
+
+  shmem_internal_get(dest, source, 4*nelems, pe);
+}
+
+
+void
+shmem_get64_nbi(void *dest, const void *source, size_t nelems, int pe)
+{
+  SHMEM_ERR_CHECK_INITIALIZED();
+
+  shmem_internal_get(dest, source, 8*nelems, pe);
+}
+
+
+void
+shmem_get128_nbi(void *dest, const void *source, size_t nelems, int pe)
+{
+  SHMEM_ERR_CHECK_INITIALIZED();
+
+  shmem_internal_get(dest, source, 16*nelems, pe);
+}
```

Comment:
@stjohnt, is there another unit test in the works that will exercise all of the NBI functions?